learning_paths/compat.py

```diff
@@ -4,14 +4,37 @@
 
 from unittest.mock import Mock
 
+from django.contrib.auth.models import AbstractBaseUser
+from opaque_keys.edx.keys import CourseKey
+
 try:
     from openedx.core.djangoapps.content.learning_sequences.api import (
         get_course_keys_with_outlines,
     )
 except ImportError:
     get_course_keys_with_outlines = Mock()
 
+try:
+    from lms.djangoapps.grades.course_grade_factory import CourseGradeFactory
+except ImportError:
+    CourseGradeFactory = Mock()
+
+try:
+    from openedx.core.djangoapps.catalog.utils import get_catalog_api_client
+except ImportError:
+    get_catalog_api_client = Mock()
```

Comment:
I think we shouldn't use mocks in code that runs on a production instance. If an import path changes in a new Open edX release, we want this to raise an exception. If it doesn't, catching the bugs will become much more complex.

Instead of global imports, we should encapsulate these imports inside of the functions. When we want to mock them for tests, we will patch the whole function, without worrying about the `ImportError`. Example:

```python
def get_catalog_api_client(user: User) -> requests.Session:
    # noinspection PyUnresolvedReferences,PyPackageRequirements
    from openedx.core.djangoapps.catalog.utils import get_catalog_api_client

    return get_catalog_api_client(user)
```

The `CourseGradeFactory` import should be in the `get_user_course_grade` function.

Reply:
I just followed what was existing in the repo previously. But you're right. I've updated them.